packages/builder/src/builder.js

```diff
@@ -541,47 +488,26 @@ export default class Builder {
         )
       }
     }
+  }
 
+  async compileTemplates(templateContext) {
     await this.nuxt.callHook('build:templates', {
-      templatesFiles,
-      templateVars,
+      ...templateContext,
       resolve: r
     })
 
     // Prepare template options
-    let lodash = null
-    const templateOptions = {
-      imports: {
-        serialize,
-        serializeFunction,
-        devalue,
-        hash,
-        r,
-        wp,
-        wChunk,
-        resolvePath: this.nuxt.resolver.resolvePath,
-        resolveAlias: this.nuxt.resolver.resolveAlias,
-        relativeToBuild: this.relativeToBuild,
-        // Legacy support: https://github.com/nuxt/nuxt.js/issues/4350
-        _: new Proxy({}, {
-          get(target, prop) {
-            if (!lodash) {
-              consola.warn('Avoid using _ inside templates')
-              lodash = require('lodash')
-            }
-            return lodash[prop]
-          }
-        })
-      },
-      interpolate: /<%=([\s\S]+?)%>/g
+    const templateOptions = templateContext.templateOptions
```

Comment:
We can use destructure syntax here :)

Reply:
I wrote it like at first, but it feels less readable and will trigger `getter` twice

```js
const templateOptions = {
 ...templateContext.templateOptions,
 imports: {
    ...templateContext.templateOptions.imports
    resolvePath: this.nuxt.resolver.resolvePath,
 }
}
```

Reply:
For destructure, good idea, will change it:

```js
const { templateOptions } = templateContext
```